java/com/google/flatbuffers/FlexBuffers.java

```diff
@@ -0,0 +1,864 @@
+                        // Convert other things to uint.
+                        return 0;
+                }
+        }
+
+        public long asLong() {
+            if (type == FBT_INT) {
+                // A fast path for the common case.
+                return readLong(bb, end, parentWidth);
+            } else
+                switch (type) {
+                    case FBT_INDIRECT_INT: return readLong(bb, indirect(bb, end, parentWidth), byteWidth);
+                    case FBT_UINT: return readUInt(bb, end, parentWidth);
+                    case FBT_INDIRECT_UINT: return readUInt(bb, indirect(bb, end, parentWidth), parentWidth);
+                    case FBT_FLOAT: return (long) readDouble(bb, end, parentWidth);
+                    case FBT_INDIRECT_FLOAT: return (long) readDouble(bb, indirect(bb, end, parentWidth), byteWidth);
+                    case FBT_NULL: return 0;
+                    case FBT_STRING: {
+                        try {
+                            return Long.parseLong(asString());
+                        }catch (NumberFormatException nfe) {
+                            return 0; //same as C++ implementation
+                        }
+                    }
+                    case FBT_VECTOR: return asVector().size();
+                    case FBT_BOOL: return readInt(bb, end, parentWidth);
+                    default:
+                        // Convert other things to int.
+                        return 0;
+                }
+        }
+
+        public double asFloat() {
+            if (type == FBT_FLOAT) {
+                // A fast path for the common case.
+                return readDouble(bb, end, parentWidth);
+            } else
+                switch (type) {
+                    case FBT_INDIRECT_FLOAT: return readDouble(bb, indirect(bb, end, parentWidth), byteWidth);
+                    case FBT_INT: return readInt(bb, end, parentWidth);
+                    case FBT_UINT:
+                    case FBT_BOOL:
+                        return readUInt(bb, end, parentWidth);
+                    case FBT_INDIRECT_INT: return readInt(bb, indirect(bb, end, parentWidth), byteWidth);
+                    case FBT_INDIRECT_UINT: return readUInt(bb, indirect(bb, end, parentWidth), byteWidth);
+                    case FBT_NULL: return 0.0;
+                    case FBT_STRING: return Double.parseDouble(asString());
+                    case FBT_VECTOR: return asVector().size();
+                    default:
+                        // Convert strings and other things to float.
+                        return 0;
+                }
+        }
+
+        public Key asKey() {
+            if (type == FBT_KEY) {
+                return new Key(bb, indirect(bb, end, parentWidth), byteWidth);
+            } else {
+                return Key.empty();
+            }
+        }
+
+        public String asString() {
+            if (type == FBT_KEY || type == FBT_STRING || type == FBT_BLOB) {
```

Comment:
Actually there is a problem indeed, but I am bit confused about your comment.

According to [FlexBuffer Internals](https://google.github.io/flatbuffers/flatbuffers_internals.html):

> Strings (TYPE_STRING) are similar to blobs, except they have an additional 0 termination byte for convenience, and they MUST be UTF-8 encoded (since an accessor in a language that does not support pointers to UTF-8 data may have to convert them to a native string type).

So actually it should work correctly for `FBT_STRING` as a 0-terminated array of bytes are expected. But not for `FBT_KEY` and FBT_BLOB`.

So the solution would be to return not-empty only `if (type == FBT_STRING) {` correct? I am missing something?

Reply:
Now I understand your point. 

As for `FBT_KEY`, if I understood correct, C++ version will return empty string, so I will try to keep the same behavior.

Reply:
Alright, I will allow `FBT_KEY` to be returned as string